common/evaluate.py

```diff
@@ -11,15 +43,44 @@ def get_meta(login):
 
 
 @django_rq.job
-def evaluate_job(s: Submit):
-    result_path = os.path.join(
-        'submit_results',
-        *s.path_parts(),
-    )
+def evaluate_job(submit_url, task_url, token, meta):
+    logging.basicConfig(level=logging.DEBUG)
+    s = requests.Session()
 
-    evaluate(
-        "tasks/{}".format(s.assignment.task.code),
-        s.dir(),
-        result_path,
-        {**get_meta(s.student.username), 'submit_id': s.id},
-    )
+    logging.info(f"Evaluating {submit_url}")
+
+    with tempfile.TemporaryDirectory() as workdir:
+        os.chdir(workdir)
```

Comment:
Nechtělo by to tady ve finally bloku vrátit chdir zpátky? Ať nám to nedělá bordel jako předtím, že se měnil workdir.

Reply:
na zacatku kazde evaluace se vytvari docasny adresar, prepiname se do nej a pak se tam nastahujou zdrojaky + task, takze by to asi melo byt OK
A myslim, ze rq nakonec provadi fork pro kazdy job